include/zephyr/drivers/stepper/stepper_trinamic.h

```diff
@@ -159,8 +179,13 @@ struct tmc_ramp_generator_data {
  * @retval -ENOSYS If not implemented by device driver
  * @retval 0 Success
  */
+#ifdef CONFIG_STEPPER_ADI_TMC5041
 int tmc5041_stepper_set_ramp(const struct device *dev,
 			     const struct tmc_ramp_generator_data *ramp_data);
+#elif CONFIG_STEPPER_ADI_TMC5160
+int tmc5160_stepper_set_ramp(const struct device *dev,
+			     const struct tmc_ramp_generator_data *ramp_data);
+#endif
```

Comment:
The `set_ramp_parameters` function is better off being within the respective drivers, i.e., tmc5041.c or tmc5160.c and then having a `set_ramp_parameters`function in `stepper.h`.

Reply:
We are gathering all trinamic specific functions in this file `stepper_trinamic.h`. IFDEFs are not really required coz compiler will report an error anyhow :)